### DISCUSSION TITLE
[v4] - Failed to construct 'WebSocket': An insecure WebSocket connection may not be initiated

### DISCUSSION THREAD

**Discussion by kempsteven** [adonisjs/core#1885] [Help]
So I implemented a websocket connection and worked perfectly in my local machine, but ones it was deployed, I am getting an
error:

```http
DOMException: Failed to construct 'WebSocket': An insecure WebSocket connection
may not be initiated from a page loaded over HTTPS.
```
Its probably because the websocket connection does not have ssl, and im connecting through a https site.
so how do I implement a secure websocket connection?

**Comment by thetutlage** [comment]
Which version of the framework are you using?
